cpp/include/raft/linalg/map.cuh

```diff
@@ -22,119 +22,74 @@
- * @tparam OutType data-type of result of type raft::device_mdspan
- * @tparam Args additional parameters
- * @param[in] handle raft::device_resources
- * @param[in] in the input of type raft::device_mdspan
- * @param[out] out the output of the map operation of type raft::device_mdspan
- * @param[in] map the device-lambda
- * @param[in] args additional input arrays
+ * @brief Map a function over zero or more input mdspans.
+ *
+ * Usage example:
+ * @code{.cpp}
+ *  auto input = raft::make_device_vector<int>(res, n);
+ *  ... fill input ..
+ *  auto squares = raft::make_device_vector<int>(res, n);
+ *  raft::linalg::map_offset(res, squares.view(), raft::sq_op{}, input.view());
+ * @endcode
+ *
+ * @tparam OutType data-type of the result (device_mdspan)
+ * @tparam Func the device-lambda performing the actual operation
+ * @tparam InTypes data-types of the inputs (device_mdspan)
+ *
+ * @param[in] res raft::device_resources
+ * @param[out] out the output of the map operation (device_mdspan)
+ * @param[in] f device lambda
+ * @param[in] ins the inputs (each of the same size as the output) (device_mdspan)
  */
-template <typename InType,
-          typename MapOp,
-          typename OutType,
-          int TPB = 256,
-          typename... Args,
-          typename = raft::enable_if_input_device_mdspan<InType>,
-          typename = raft::enable_if_output_device_mdspan<OutType>>
-void map(raft::device_resources const& handle, InType in, OutType out, MapOp map, Args... args)
+template <typename OutType,
+          typename Func,
+          typename... InTypes,
+          typename = raft::enable_if_output_device_mdspan<OutType>,
+          typename = raft::enable_if_input_device_mdspan<InTypes...>>
+void map(const raft::device_resources& res, OutType out, Func f, InTypes... ins)
 {
-  using in_value_t  = typename InType::value_type;
-  using out_value_t = typename OutType::value_type;
-
-  RAFT_EXPECTS(raft::is_row_or_column_major(out), "Output is not exhaustive");
-  RAFT_EXPECTS(raft::is_row_or_column_major(in), "Input is not exhaustive");
-  RAFT_EXPECTS(out.size() == in.size(), "Size mismatch between Input and Output");
-
-  if (out.size() <= std::numeric_limits<std::uint32_t>::max()) {
-    map_k<in_value_t, MapOp, std::uint32_t, TPB, out_value_t, Args...>(
-      out.data_handle(), out.size(), map, handle.get_stream(), in.data_handle(), args...);
-  } else {
-    map_k<in_value_t, MapOp, std::uint64_t, TPB, out_value_t, Args...>(
-      out.data_handle(), out.size(), map, handle.get_stream(), in.data_handle(), args...);
-  }
+  return detail::map<false>(res, out, f, ins...);
 }
 
 /**
- * @brief Perform an element-wise unary operation on the input offset into the output array
+ *  @brief Map a function over zero-based flat index (element offset) and zero or more inputs.
  *
  * Usage example:
  * @code{.cpp}
- *  #include <raft/core/device_mdarray.hpp>
- *  #include <raft/core/handle.hpp>
- *  #include <raft/core/operators.hpp>
```

Comment:
Why remove these from the usage example? The goal with these examples is to make them copy/paste for the users so they don't have to go searching through headers to find what they need.